src/map/battlefield.cpp

```diff
@@ -453,11 +540,13 @@ void CBattlefield::OpenChestinBcnm() {
 
 bool CBattlefield::loseBcnm() {
     beforeCleanup();
-    for (auto&& PChar : m_PlayerList)
+    for (int i = 0; i < m_PlayerList.size(); i++)
```

Comment:
ah right - the previous code didn't remove them, though, which is why I think this is wrong (and the analogous winBcnm doesn't either) - probably needs a check for bcnm->lost in onEventFinish to mirror how win works